src/main/java/net/dv8tion/jda/api/entities/ThreadChannel.java

```diff
@@ -71,119 +136,319 @@ default GuildMessageChannel getParentMessageChannel()
+     * <br>
+     * This will be null if the member is not cached.
+     *
+     * @see #getThreadMemberById(long)
+     *
+     * @return The owner of this thread as a {@link ThreadMember}.
+     */
     @Nullable
     default ThreadMember getOwnerThreadMember()
     {
         return getThreadMemberById(getOwnerIdLong());
     }
 
+    /**
+     * Whether this thread has been archived.
+     *
+     * This method will consider locked channels to also be archived.
+     *
+     * @see #isLocked()
+     * @return true if this thread has been archived, false otherwise.
+     */
     boolean isArchived();
 
+    /**
+     * The last updated time of the archive info of this thread.
+     *
+     * <p>This timestamp will be updated when any of the following happen:
+     * <ul>
+     *     <li>The channel is archived</li>
+     *     <li>The channel is unarchived</li>
+     *     <li>The AUTO_ARCHIVE_DURATION is changed.</li>
+     * </ul>
+     *
+     * @see ChannelField#ARCHIVED_TIMESTAMP
+     *
+     * @return the time of the last archive info update.
+     */
     OffsetDateTime getTimeArchiveInfoLastModified();
 
+    /**
+     * The inactivity timeout of this thread.
+     *
+     * If a message is not sent within this amount of time, the thread will be automatically archived.
+     *
+     * A thread archived this way can be unarchived by any member.
+     *
+     * @return the time before which a thread will automatically be archived.
+     */
     @Nonnull
     AutoArchiveDuration getAutoArchiveDuration();
 
+
+
+
+    /**
+     * The slowmode time of this thread. This determines the time each non-moderator must wait before sending another message.
+     *
+     * @see net.dv8tion.jda.api.managers.channel.concrete.ThreadChannelManager#setSlowmode(int)
+     *
+     * @return The time between a member sending two messages.
+     */
     int getSlowmode();
 
+    /**
+     * Joins this thread, adding the current account to the member list of this thread.
+     *
+     * Note that joining threads is not a requirement of getting events about the thread.
+     *
+     * <br>This will have no effect if the current account is already a member of this thread.
+     *
+     * @throws IllegalStateException
+     *         If this thread is locked or archived.
+     *
+     * @return {@link RestAction}
+     */
     @CheckReturnValue
     RestAction<Void> join();
 
+    /**
+     * Leaves this thread, removing the current account from the member list of this thread.
+     *
+     * <br>This will have no effect if the current account is not a member of this thread.
+     *
+     * @throws IllegalStateException
+     *         If this thread is locked or archived.
+     *
+     * @return {@link RestAction}
+     */
     @CheckReturnValue
     RestAction<Void> leave();
 
+    //TODO-hex: do the permission requirements differ for private threads?
+    //this is probably also affected by private threads that are not invitable
```

Comment:
Yeah this definitely needs to check for moderator/thread-owner when private. The `impl` code needs to be updated to do so.

Reply:
We also need to determine what happens if you try to add a thread member who doesn't have access to the parent channel in the public scenario too.